### ISSUE TITLE
Add Mermaid/SVG diagrams for folder structure changes

### ISSUE THREAD

**Issue by dustinkeeton** [dustinkeeton/obsidian-synapse#48]
## Problem / Motivation

After operations like **deep dive** and **organize**, the vault's folder structure can change significantly — new folders are created, notes are moved, and tree hierarchies emerge. Users currently have no visual summary of what changed. They're left scanning the file explorer to piece together the before/after state, which is tedious for large runs (deep dive can generate 50+ notes, organize can move dozens of files).

A visual diagram would give users an immediate, intuitive understanding of what Auto Notes did to their vault.

## Proposed Solution

After structure-altering operations complete, generate a **Mermaid diagram** (rendered natively by Obsidian) embedded in a summary note. Optionally export an **SVG** snapshot for users who want a static image.

### Diagram types

**Deep dive tree diagram** — shows the generated topic tree rooted at the source note:

````markdown
```mermaid
graph TD
    ML["Machine Learning"] --> NN["Neural Networks"]
    ML --> GD["Gradient Descent"]
    NN --> BP["Backpropagation"]
    NN --> AF["Activation Functions"]
    GD --> LR["Learning Rate Schedules"]
```
````

**Organize move diagram** — shows before/after folder paths as a flowchart:

````markdown
```mermaid
graph LR
    subgraph Before
        A["notes/neural-networks.md"]
        B["notes/gradient-descent.md"]
    end
    subgraph After
        C["AI & ML/Neural Networks/neural-networks.md"]
        D["AI & ML/Optimization/gradient-descent.md"]
    end
    A --> C
    B --> D
```
````

### Where diagrams live

- **Deep dive:** embed the tree diagram in the syllabus index note (see #42) — natural home since the syllabus already shows the topic outline
- **Organize:** generate a summary note at `.auto-notes/organize/summaries/{date}-organize-summary.md` with the move diagram, and show a notice linking to it
- Both diagram types should also be viewable from the sidebar panel

### SVG export (stretch goal)

Optionally render the Mermaid diagram to SVG using `mermaid-js` at generation time and save alongside the summary note. This is useful for users who want to share or archive the diagram outside Obsidian.

## Sub-issues

- [ ] Build a Mermaid diagram generator utility (`src/shared/diagram-generator.ts`) that accepts a tree or move-list and outputs a Mermaid code block string
- [ ] Generate deep dive tree diagrams from `DeepDiveRun` proposal tree and embed in syllabus note (#42)
- [ ] Generate organize move diagrams from `OrganizeMoveSnapshot` records and write to a summary note
- [ ] Surface diagram links in sidebar panel and post-operation notices
- [ ] (Stretch) Add SVG export using Mermaid's `renderAsync` API

## Context

- Related to #42 (syllabus navigation) — the deep dive tree diagram is a natural addition to the syllabus index note
- Deep dive tree data: `DeepDiveRun.proposalIds`, `DeepDiveProposal.childProposalIds`, `depth` (`src/deep-dive/types.ts`)
- Organize move data: `OrganizeMoveSnapshot.originalPath`, `currentPath` (`src/organize/types.ts`)
- Obsidian renders Mermaid code blocks natively — no plugin dependency needed
